File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.java
class Solution {
    public String removeOuterParentheses(String s) {
        int a=0;
        
        String ans="";
        
        for(int i=0; i<s.length(); i++){
            char ch=s.charAt(i);
            
            if(ch=='('){
                if(a!=0){
                    ans+=ch;
                }
                
                a++;
            } else {
                if(a!=1){
                    ans+=ch;
                }
                
                a--;
            }
        }
        
        return ans;
    }
}

File: 1021-remove-outermost-parentheses/1021-remove-outermost-parentheses.cpp
class Solution {
public:
    string removeOuterParentheses(string s) {
        string ans="";
        int a=0;
        
        for(int i=0; i<s.size(); i++){
            if(s[i]=='('){
                a++;
                if(a!=1){
                    ans+=s[i];
                }
            } else {
                a--;
                if(a!=0){
                    ans+=s[i];
                }
            }
        }
        
        return ans;
    }
};